### PR TITLE
Support for GDAL Integer64 #35 and #40 Support for PostgreSQL 9.5 beta1

### PR DESCRIPTION
**Pull request by robe2**
This is to fix issue #35.  See what happens when you don't fix things.  I have to fix them myself.

### CODE REVIEW

File: ogr_fdw.c [hunk 1]
```diff
@@ -1038,6 +1054,9 @@ ogrFeatureToSlot(OGRFeatureH feat, TupleTableSlot *slot, TupleDesc tupdesc)
 						break;
 					}
 					case OFTInteger:
+					#if GDAL_VERSION_MAJOR >= 2
+                        case OFTInteger64:
```
Comment (pramsey) [inline ogr_fdw.c]: This is #ifdef'ed against the GDAL version in ogr_fdw_info.c, it should be here too

File: ogr_fdw.c [hunk 2]
```diff
@@ -583,11 +583,27 @@ ogrGetForeignPlan(PlannerInfo *root,
 	ogrFinishConnection(&(planstate->ogr));
 	
 	/* Create the ForeignScan node */
+	
+/*
+ * Require PostgreSQL >= 9.5
+ */
+#if PG_VERSION_NUM >= 90500
```
Comment (pramsey) [inline ogr_fdw.c]: with clever formatting you can narrow this #ifdef to being just around the new 9.5 parameter

**Comment by robe2** [comment]
Is it okay now.  I made all the changes you requested teacher.

File: ogr_fdw.c [hunk 3]
```diff
@@ -583,11 +583,24 @@ ogrGetForeignPlan(PlannerInfo *root,
 	ogrFinishConnection(&(planstate->ogr));
 	
 	/* Create the ForeignScan node */
+	
+
```
Comment (pramsey) [inline ogr_fdw.c]: hah! whitespace noise! unacceptable!

File: ogr_fdw.c [hunk 4]
```diff
@@ -1038,6 +1051,9 @@ ogrFeatureToSlot(OGRFeatureH feat, TupleTableSlot *slot, TupleDesc tupdesc)
 						break;
 					}
 					case OFTInteger:
+#if GDAL_VERSION_MAJOR >= 2
+                    case OFTInteger64:
```
Comment (pramsey) [inline ogr_fdw.c]: match the spacing, not soft tabs, but real ones please

File: ogr_fdw_info.c [hunk 5]
```diff
@@ -265,6 +265,12 @@ ogrGenerateSQL(const char *source, const char *layer)
 			case OFTStringList:
 				printf("varchar[]");
 				break;
+				
+#if GDAL_VERSION_MAJOR >= 2
+		   case OFTInteger64:
```
Comment (pramsey) [inline ogr_fdw_info.c]: also here, real tabs please

**Comment by robe2** [comment]
Oh thank you so much.  I'm so honored to get my patch accepted. I'm encouraged to submit more patches.  I'll maybe start small by trying to fix those dangling unused variables you left behind and in doing so practice the art of spacing.  Then dream bigger like implementing the import foreign schema thing. for 9.5

**Pull request merged**